pkg/operator/controller/dns_status.go

```diff
@@ -25,6 +30,11 @@ func (r *reconciler) syncDNSStatus(dns *operatorv1.DNS, clusterIP, clusterDomain
 	updated.Status.ClusterDomain = clusterDomain
 	updated.Status.Conditions = computeDNSStatusConditions(dns, clusterIP, haveDNSDaemonset, dnsDaemonset, haveNodeResolverDaemonset, nodeResolverDaemonset)
 	if !dnsStatusesEqual(updated.Status, dns.Status) {
+		// skip frequent updates to progressing and degraded conditions to reduce status flapping.
+		if conditionsUpdatedWithinLast(dns.Status.Conditions, conditionUpdateInterval) {
```

Comment:
It cannot leave the condition at `Progressing=True` because it prevents updates only when `Progressing=False`. The only time we will miss updates is when `Progressing` becomes `True` temporarily, which should be ok.